### PR TITLE
Qute message bundles - add @MessageParam

### PR DESCRIPTION
**Pull request by mkouba**
- to bind an explicit name for a message bundle method parameter

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
Awesome!

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java [hunk 1]
```diff
@@ -552,7 +560,8 @@ private String generateImplementation(ClassInfo defaultBundleInterface, String d
 
             if (messageAnnotation == null) {
                 throw new MessageBundleException(
-                        "A message bundle interface method must be annotated with @Message: " + method);
+                        String.format("A message bundle interface method must be annotated with @Message on %s: ",
```
Comment (gsmet) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:563]: I think there's a `%s` missing here or the method has one too many parameters.

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java [hunk 2]
```diff
@@ -618,6 +623,27 @@ private String generateImplementation(ClassInfo defaultBundleInterface, String d
...
+            AnnotationValue paramAnnotationValue = paramAnnotation.value();
+            if (paramAnnotationValue != null && !paramAnnotationValue.asString().equals(Message.ELEMENT_NAME)) {
+                name = paramAnnotationValue.asString();
+            }
+        }
+        if (name == null) {
+            throw new MessageBundleException("Unable to determine the name of the parameter at position " + position
+                    + " in method "
+                    + method.declaringClass().name() + "#" + method.name()
+                    + "() - compile the class with debug info enabled (-g) or parameter names recorded (-parameters), or use @MessageParam to specify the value");
```
Comment (gsmet) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:642]: Should the `-parameters` recommendation come first?

**Review by gsmet — DISMISSED** [review]
Spotted an issue and added a small suggestion (that can be ignored if not appropriate).

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java (hunk 1 above)
Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:563]: Good catch!

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java (hunk 2 above)
Comment (mkouba) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:642]: Well, it depends. From jandex POV it does not matter. Also maven compiler does `-g` by default so I don't expect this message to be logged very often.

**Comment by mkouba** [comment]
@gsmet I think it's good to go now...

**Pull request merged**